File: dataset_reduzido/codenet/Java/Code/s818271374.java
import java.util.*;
public class Main {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int N = sc.nextInt();
    int count1 = 0;
    int count2 = 0;
    for(int i=0;i<N;i++){
      int a = sc.nextInt();
      if(a%4==0){
        count1++;
      }else if(a%2==0 && (a/2)%2==1){
        count2++;
      }
    }
    if(count1>=N/2){
      System.out.println("Yes");
    }else if(((N/2-count1)*2+N%2)==count2){
      System.out.println("Yes");
    }else{
      System.out.println("No");
    }
  }
}

File: output/Llama-2-100/codenet/Java/Python/s818271374.py
def main():
    n = int(input())
    count1 = 0
    count2 = 0
    for i in range(n):
        a = int(input())
        if a%4==0:
            count1 += 1
        elif a%2==0 and (a/2)%2==1:
            count2 += 1
    if count1>=n/2:
        print("Yes")
    elif (n/2-count1)*2+n%2==count2:
        print("Yes")
    else:
        print("No")
